### PR TITLE
[Fix] Add padding for odd embedding dimensions in tensors (sparse encoders)

### PR DESCRIPTION
**Pull request by jadermcs**
Fix pairwise_angle_sim to handle odd embedding dimensions

The function uses `torch.chunk(x, 2, dim=1)` to split embeddings into real and imaginary components for complex angle calculations. This fails when the embedding dimension is odd since it cannot be evenly divided by 2.
```python
# y.size() == (32, 30503) (batch, vocabulary size, e.g., splade)

a, b = torch.chunk(x, 2, dim=1)
c, d = torch.chunk(y, 2, dim=1)
# c.size() == (32, 15252)
# d.size() == (32, 15251) ^^^ mismatch
z = torch.sum(c**2 + d**2, dim=1, keepdim=True) # size of c and d mismatches returning ERROR
```

I added a check for odd embedding dimensions and applied zero-padding (`torch.nn.functional.pad`) to both x and y tensors before chunking. The padding adds a single zero to the right side of the embedding dimension, which has minimal impact on the angle similarity computation while ensuring `torch.chunk` can split the tensors evenly.

### CODE REVIEW

**Comment by tomaarsen** [comment]
Hello!

Nice work, I totally agree that we should patch this. I added a test and ran `pre-commit` so the quality checks are happy.
I'll merge this once the tests go green. Thanks!

- Tom Aarsen

**Comment by tomaarsen** [comment]
Looks like there's some unrelated timeouts. No worries.

- Tom Aarsen
